apps/meteor/server/models/raw/Messages.ts

```diff
@@ -1649,4 +1649,13 @@ export class MessagesRaw extends BaseRaw<IMessage> implements IMessagesModel {
 		};
 		return this.updateOne(query, update);
 	}
+
+	findVisitorLastMessageByRoomId(rid: string): Promise<IMessage | null> {
+		const query = {
+			rid,
+			token: { $exists: true },
```

Comment:
:hmmm: not sure if we need a `rid + token` index for this

Reply:
You're right, I believe we'll need this index. However considering this is Messages collection which is always growing rapidly, I'd like to be cautious and not add this index because this is only needed for a single feature and majority of the application would not use this index. So let me try exploring other ways we can fix this issue